featuremanagement/_models/_variant_assignment_reason.py

```diff
@@ -0,0 +1,19 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+from enum import Enum
+
+
+class VaraintAssignmentReason(Enum):
```

Comment:
Would you mind adding some comments to explain each assignment reason?

You can refer to the .NET feature management.

/// <summary>
/// Variant allocation did not happend. No variant is assigned.
/// </summary>
None,

/// <summary>
/// The default variant is assigned when a feature flag is disabled.
/// </summary>
DefaultWhenDisabled,

/// <summary>
/// The default variant is assigned because of no applicable user/group/percentile allocation when a feature flag is enabled.
/// </summary>
DefaultWhenEnabled,

/// <summary>
/// The variant is assigned because of the user allocation when a feature flag is enabled.
/// </summary>
User,

/// <summary>
/// The variant is assigned because of the group allocation when a feature flag is enabled.
/// </summary>
Group,

/// <summary>
/// The variant is assigned because of the percentile allocation when a feature flag is enabled.
/// </summary>
Percentile